### PR TITLE
Muhimbura Brian Assesment

### PR DESCRIPTION
**Pull request by BrianMuhimbura**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by galovics** [comment]
Please update your PR with the latest develop changes and make sure you use rebase (I see you did a merge).
Also, make sure you're squashing your commits for the PRs. Thanks.

**Comment by galovics** [comment]
@BrianMuhimbura can you please link the related JIRA ticket to this?

File: fineract-db/V1__MyInterviewMigration.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE `m_clients`
```
Comment (galovics) [inline fineract-db/V1__MyInterviewMigration.sql]: This is definitely wrong cause Fineract is using Liquibase, not flyway.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/build.gradle [hunk 2]
```diff
@@ -1,4 +1,5 @@
-/**
+targetCompatibility = JavaVersion.VERSION_16
```
Comment (adamsaghy) [inline fineract-provider/build.gradle:1]: Please remove this! In the top level build.gradle, it is already configured to marke compatibility to JavaVersion.VERSION_17

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java [hunk 3]
```diff
@@ -6,9 +6,9 @@
  * to you under the Apache License, Version 2.0 (the
  * "License"); you may not use this file except in compliance
  * with the License. You may obtain a copy of the License at
- *
+ * <p>
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java:9]: Please remove these <p> tags

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientApiConstants.java [hunk 4]
```diff
@@ -85,6 +85,7 @@ public class ClientApiConstants {
     public static final String firstnameParamName = "firstname";
     public static final String middlenameParamName = "middlename";
     public static final String lastnameParamName = "lastname";
+    public static final String surnameParamName = "surname";
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientApiConstants.java:88]: lastname and surname is equivalent, no?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java [hunk 5]
```diff
@@ -101,6 +102,9 @@ public final class Client extends AbstractPersistableCustom {
     @Column(name = "lastname", length = 50, nullable = true)
     private String lastname;
 
+    @Column(name = "surname", length = 50, nullable = true)
+    private String surname;
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java:106]: Why do we need surname? We already have lastname field

File: fineract-provider/src/main/java/org/apache/fineract/useradministration/domain/AppUser.java [hunk 6]
```diff
@@ -74,6 +74,9 @@ public class AppUser extends AbstractPersistableCustom implements PlatformUser {
     @Column(name = "lastname", nullable = false, length = 100)
     private String lastname;
 
+    @Column(name = "gender", nullable = false, length = 100)
+    private Integer gender;
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/useradministration/domain/AppUser.java:78]: What is the reason to add gender to Appuser? Seems odd to me...

File: fineract-provider/src/main/resources/db/changelog/tenant/changelog-tenant.xml [hunk 7]
```diff
@@ -39,4 +39,5 @@
     <include file="parts/0017_fix_stretchy_reports.xml" relativeToChangelogFile="true"/>
     <include file="parts/0018_pentaho_reports_to_table.xml" relativeToChangelogFile="true"/>
     <include file="parts/0019_refactor_loan_transaction.xml" relativeToChangelogFile="true"/>
+    <include file="parts/0020_MyInterviewMigration.xml" relativeToChangelogFile="true"/>
```
Comment (adamsaghy) [inline fineract-provider/src/main/resources/db/changelog/tenant/changelog-tenant.xml:42]: Interview?? is this an interview task?

**Review by adamsaghy — CHANGES_REQUESTED** [review]
@BrianMuhimbura Can you help me to understand better this PR? What is this for?

**Comment by galovics** [comment]
No reaction after ~20 days, closing.

File: fineract-provider/src/main/resources/db/changelog/tenant/changelog-tenant.xml (hunk 7 above)
Comment (BrianMuhimbura) [reply fineract-provider/src/main/resources/db/changelog/tenant/changelog-tenant.xml:42]: Hello @adamsaghy this PR was an interview task

**Pull request closed without merging**